mne/io/base.py

```diff
@@ -363,6 +368,24 @@ def __init__(self, info, preload=False,
         self._comp = None
         self._filenames = list(filenames)
         self.orig_format = orig_format
+        # Sanity check and set original units, if provided by the reader:
+        if orig_units:
+            ch_l = list(info['ch_names'])
+
+            # STI 014 channel is native only to fif ... for all other formats
+            # this was artificially added by the IO procedure
+            if 'STI 014' in ch_l and not self.filenames[0].endswith('.fif'):
+                ch_l.remove('STI 014')
+
+            # check that channels correspond. note that long ch_names get
+            # truncated, so check first 15 characters
+            orig_units_chs = [ch[:15] for ch in orig_units.keys()]
```

Comment:
no need for .keys()

Reply:
it seems you want to do an intersection between set of ch_l and orig_units. why not using a `set` and use its builtin intersection method?

https://docs.python.org/3/tutorial/datastructures.html#sets
https://docs.python.org/3/library/stdtypes.html#set

Reply:
the `ch[:15]` part is important to account for truncated channel names, so I need the first list comprehension for this operation ... and then it's just one other line with a list comprehension --> is there a significant benefit to using a set intersection here? Or is it a question of style?